test/coe.lean

```diff
@@ -0,0 +1,44 @@
+import Std.Tactic.CoeExt
+import Std.Tactic.GuardMsgs
+
+set_option linter.missingDocs false
+
+structure WrappedNat where
+  val : Nat
+
+
+structure WrappedFun where
+  fn : Nat → Nat
+
+
+structure WrappedType where
+  typ : Type
+
+attribute [coe] WrappedNat.val
+instance : Coe WrappedNat Nat where coe := WrappedNat.val
+
+#eval Std.Tactic.Coe.registerCoercion ``WrappedFun.fn (some ⟨1, 0, .coeFun⟩)
+instance : CoeFun WrappedFun (fun _ => Nat → Nat) where coe := WrappedFun.fn
+
+#eval Std.Tactic.Coe.registerCoercion ``WrappedType.typ (some ⟨1, 0, .coeSort⟩)
+instance : CoeSort WrappedType Type where coe := WrappedType.typ
```

Comment:
These could potentially be spelt `@[coe coeFun]` and `@[coe coeSort]`

Reply:
... but I'd strongly prefer to declare that out of scope, Mathlib already uses this approach:

https://github.com/leanprover-community/mathlib4/blob/15ce6b63385ff165118b075a86a21a9bfabb4664/Mathlib/Data/FunLike/Basic.lean#L152-L154